doc/foreshadow_class.uml

```diff
@@ -0,0 +1,169 @@
+@startuml
+
+skinparam BackgroundColor transparent
+skinparam Shadowing false
+
+' Diagram setup
+hide empty members
+left to right direction
+set namespaceSeparator none
+
+package foreshadow.utils {
+    class check_df << (M,lemonchiffon) >>
+}
+package foreshadow.logging {
+    class ForeshadowLogger
+}
+package foreshadow.intents {
+    abstract class BaseIntent {
+        list engineering_pipeline
+        list preprocessing_pipeline
+        resolve_intent()
+    }
+
+    class DropIntent
+    class NumericalIntent
+    class CategoricalIntent
+    class TextIntent
+
+    BaseIntent <|-- DropIntent
+    BaseIntent <|-- NumericalIntent
+    BaseIntent <|-- CategoricalIntent
+    BaseIntent <|-- TextIntent
+
+    note "Config for intentless transformations (cleaner) are placed in\nBaseIntent's specification" as N1
+}
+
+together {
+    package foreshadow.transformers.core {
+        abstract class SmartTransformer {
+            bool fixed_fit
+            log_decision()
+        }
+
+        class ParallelProcessor
+        class SigCopy
+        class DropFeature
+
+        class wrap_transformer << (M,lemonchiffon) >>
+        wrap_transformer o-- SigCopy
+    }
+
+    package foreshadow.transformers.smart {
+        SmartTransformer <|-- Cleaner
+        SmartTransformer <|-- Engineerer
+        SmartTransformer <|-- Scaler
+        SmartTransformer <|-- Imputer
+        SmartTransformer <|-- CategoricalEncoder
+        SmartTransformer <|-- TextEncoder
+        SmartTransformer <|-- Reducer
+    }
+
+    package foreshadow.transformers.internal {
```

Comment:
Can we fix the line paths and ordering here to make it easier to see which internal transformers relate to which _SmartTransformers_.
This might help:
https://stackoverflow.com/questions/48712801/how-to-correct-plantuml-line-path

Reply:
I don't think this is possible in that way. I've tried using hidden connections but unfortunately, it doesn't allow for the elegant positioning of packages.

Reply:
I made a few changes that might help though.